kinto/core/cache/postgresql/__init__.py

```diff
@@ -157,10 +158,24 @@ def set(self, key, value, ttl=None):
         SELECT :key, :value, sec2ttl(:ttl)
         WHERE NOT EXISTS (SELECT * FROM upsert)
         """
+        # Query for PostgreSQL 9.4
+        query_pg_94 = """
+        WITH upsert AS (
+            UPDATE cache SET value = :value, ttl = sec2ttl(:ttl)
+             WHERE key = :key
+            RETURNING *)
+        INSERT INTO cache (key, value, ttl)
+        SELECT :key, :value, sec2ttl(:ttl)
+        WHERE NOT EXISTS (SELECT * FROM upsert);
+        """
         value = json.dumps(value)
+        params = dict(key=self.prefix + key, value=value, ttl=ttl)
         with self.client.connect() as conn:
-            conn.execute(query, dict(key=self.prefix + key,
-                                     value=value, ttl=ttl))
+            server_version = conn.connection().dialect.server_version_info
```

Comment:
**nitpick:** here https://github.com/zzzeek/sqlalchemy/blob/bc4a96836d5bfb911da26f7e77119f3a7b356f2e/lib/sqlalchemy/testing/exclusions.py#L384 they use `getattr` so maybe we should use it too and compare `version >= (9, 5)`